### ISSUE TITLE
Provide UI for configuring devappserver storage path location(s)

### ISSUE THREAD

**Issue by briandealwis** [GoogleCloudPlatform/google-cloud-eclipse#1404]
In #1245, we add support for identifying if two devappservers have conflicting resource requirements, such as the server port, admin port, and the base storage location.

The base storage path normally contains the files from the various App Engine services (datastore, blobstore, search index, and logs).  This base storage paths is set with `--storage_path` and defaults to `$TMPDIR/appengine.${APPID}.${user.name}`.  The  _APPID_ is specified with the `-A` flag, and defaults to _None_.  The individual stores can also have explicit paths set too, which can be useful for debugging purposes such as to test with canned data.

We could make the storage paths not conflict by allowing the user to either:
  - specify an _APPID_
  - specify the base storage path (passed to `--storage_path`)
  - specify the various explicit storage paths (`—blobstore_path`, `--datastore_path`, `--logs_path`, `--prospective_search_path`, `--search_indexes_path`)
It might even make sense to set the default to be kept in the per-Server directory so that the results persist between server restarts.

**Comment by briandealwis** [comment]
User request to [make this kind of thing configurable](https://stackoverflow.com/questions/44870535/how-can-i-set-the-datastore-path-in-appengine-eclipse-pydev).

**Comment by chanseokoh** [comment]
The main motivation for setting a "storage path", as I see it, is to be able to persist datastore and indexes across server runs. (As such, I think there is little value to have UI for some of the advanced options above, that's not the main point I'm about to talk about.)

The story around the datastore path and the index path seems different between dev_appserver1 and dev_appserver2. Although I don't fully understand the justification, it has been suggested that the option `--generated-dir` for dev_appserver1 shouldn't be really exposed to users and that `datastore-index-auto.xml` and `local_db.bin` that dev_appserver1 generates should be put under users' source (e.g., `src/main/webapp/WEB-INF/appengine-generated`), whether by running dev_appserver1 directly against users' webapp source directory (which I think is not a good idea considering build tools can do extra processing, or by copying the two files back to users' webapp source when dev_appserver1 terminates. That way the datastore and the auto-generated indexes would survive after each server restart.

FYI, on the IntelliJ, they decided to simply not do anything to mitigate the datastore persistence issue and not expose `--generated-dir`, as any extra attempt to mitigate the issue seems to have too many downsides. @patflynn @etanshaul

In any case, the options will be very different between dev_appserver1 and dev_appserver2, and considering that we're not yet 100% sure if it makes sense to expose `--generated-dir` or not, I think we need to discuss further before attempting to implement a UI.

**Comment by briandealwis** [comment]
Someone mentioned that it could be possible to expose the choice of devappserver1 vs 2 to the user.  We could have a combo box that allows certain fields, following the same logic with `LocalAppEngineServerLaunchConfigurationDelegate.DEV_APPSERVER2`.

**Comment by etanshaul** [comment]
We dinstinctly decided against this for IntelliJ (v1 / v2 chooser) after
much discussion. I'm trying to recall the specifics, but I know there were
a lot of complications with this and confusion for the user.

**Comment by briandealwis** [comment]
A request to [be able to specify `—skip_sdk_update_check`](https://stackoverflow.com/questions/44134717/why-is-an-exception-thrown-when-trying-to-check-for-updates-to-the-google-app-en).

**Comment by chanseokoh** [comment]
> A request to [be able to specify `--skip_sdk_update_check`](https://stackoverflow.com/questions/44134717/why-is-an-exception-thrown-when-trying-to-check-for-updates-to-the-google-app-en).

It's not clear if the user is actually using CT4E. (He says he's doing _Run As_ > _Web Application_.) It's kind of weird the debugger respond to an exception thrown from the server. Maybe he's executing the dev server manually.

In any case, if we want this `--skip_sdk_update_check`, I think we need to change the title of this issue. However, to break thing down, creating a new issue seems better.

**Comment by briandealwis** [comment]
You're right @chanseokoh: I'm mixing issues.  And given your analysis,, this specific issue should wait until we move to devappserver2.

W.r.t. the `skip_sdk_update_check` issue,  I suspect that they're using GPE as `appengine-plugins-core` support for devappserver1 always specifies `—disable_update_check`.  That said, it's not surprising it's being caught by the user's debugger: `SocketException` is a subclass of `IOException`, so the user likely has has an IOException exception breakpoint; and debugging a web-module is done by launching it within the dev app server.

**Comment by patflynn** [comment]
We want to hold off on exposing devappserver2 to users until we have a
devappserver2 release candidate that we know works for most users.

* b/hotlists/449061*
